iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheck.java

```diff
@@ -47,18 +51,61 @@ void registerObjectCheck() {
     }
   }
 
-  private static void checkAndReport(BlockObject blockObject, String message) {
+  @Override
+  Class<ServiceAccount> getGlobalResourceType() {
+    return ServiceAccount.class;
+  }
+
+  private void checkAndReport(BlockObject blockObject, String message) {
     var specAsBlockObject = blockObject.block("spec");
     var specAsAttributeObject = blockObject.attribute("spec");
     if (specAsAttributeObject.tree != null && isContainersPresentInSpecBlock(specAsBlockObject)) {
       var tokenAttribute = specAsBlockObject.attribute(KEY);
-      tokenAttribute.reportIfAbsent(specAsAttributeObject.tree.key(), message);
-      tokenAttribute.reportIfValue(isSet().negate(), message);
-      tokenAttribute.reportIfValue(isTrue(), message);
+      if (!tokenAttribute.isAbsent()) {
+        tokenAttribute.reportIfValue(TreePredicates.isSet().negate(), message);
+        tokenAttribute.reportIfValue(isTrue(), message);
+        return;
+      }
+
+      List<ServiceAccount> linkedServiceAccounts = retrieveLinkedServiceAccount(blockObject);
+      if (linkedServiceAccounts.isEmpty()) {
+        specAsAttributeObject.reportOnKey(message);
+      } else {
+        boolean hasAtLeastOneAccountCompliant = linkedServiceAccounts.stream().anyMatch(account -> account.automountServiceAccountToken().isFalse());
+        if (!hasAtLeastOneAccountCompliant) {
+          // report on first linked account - there should be only one
```

Comment:
Maybe, a combination of `filter` + `findFirst` + `isPresent` could be better here, than `anyMatch` + `get(0)`?